lib/auth/access_test.go

```diff
@@ -173,6 +178,774 @@ func TestUpsertDeleteDependentRoles(t *testing.T) {
+		wantErr      require.ErrorAssertionFunc
+		want         types.Role
+	}{
+		{
+			name:       "create role",
+			userRole:   newRole(t, "urole", nil, rcWithRoleRule(types.VerbCreate), types.RoleConditions{}),
+			createRole: newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr:    require.NoError,
+			want:       newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+		},
+		{
+			name:       "create role denied",
+			userRole:   newRole(t, "urole", nil, types.RoleConditions{}, types.RoleConditions{}),
+			createRole: newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+		{
+			name:         "create role collision",
+			userRole:     newRole(t, "urole", nil, rcWithRoleRule(types.VerbCreate), types.RoleConditions{}),
+			startingRole: newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			createRole:   newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAlreadyExists(err))
+			},
+		},
+		{
+			name:         "create role collision with access denied",
+			userRole:     newRole(t, "urole", nil, types.RoleConditions{}, types.RoleConditions{}),
+			startingRole: newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			createRole:   newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+		{
+			name: "create role where clause",
+			userRole: newRole(t, "urole", nil, rcWithRoleRuleWhere([]string{types.VerbCreate},
+				builder.Equals(
+					builder.Identifier(`resource.metadata.labels["label"]`),
+					builder.String("value"),
+				).String(),
+			), types.RoleConditions{}),
+			createRole: newRole(t, "create", map[string]string{
+				"label": "value",
+			}, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: require.NoError,
+			want: newRole(t, "create", map[string]string{
+				"label": "value",
+			}, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+		},
+		{
+			name: "create role where clause denied",
+			userRole: newRole(t, "urole", nil, rcWithRoleRuleWhere([]string{types.VerbCreate},
+				builder.Equals(
+					builder.Identifier(`resource.metadata.labels["label"]`),
+					builder.String("value"),
+				).String(),
+			), types.RoleConditions{}),
+			createRole: newRole(t, "create", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
```

Comment:
did you mean to parallel the subtests?

Reply:
Yes I did. I remember thinking of needing to go back and do this, but I forgot!